libs/backend/src/cast_vote_records/import.test.ts

```diff
@@ -1,7 +1,325 @@
+  fs.appendFileSync(
+    path.join(exportDirectoryPath, CastVoteRecordExportFileName.METADATA),
+    '\n'
+  );
+
+  expect(await readCastVoteRecordExport(exportDirectoryPath)).toEqual(
+    err({ type: 'authentication-error' })
+  );
+});
+
+test('metadata file not found during import', async () => {
+  mockFeatureFlagger.enableFeatureFlag(
+    BooleanEnvironmentVariableName.SKIP_CAST_VOTE_RECORDS_AUTHENTICATION
+  );
+  const exportDirectoryPath = castVoteRecordExport.asDirectoryPath();
+  fs.rmSync(
+    path.join(exportDirectoryPath, CastVoteRecordExportFileName.METADATA)
+  );
+
+  expect(await readCastVoteRecordExport(exportDirectoryPath)).toEqual(
+    err({ type: 'metadata-file-not-found' })
+  );
+});
+
+test('metadata file parse error during import', async () => {
+  mockFeatureFlagger.enableFeatureFlag(
+    BooleanEnvironmentVariableName.SKIP_CAST_VOTE_RECORDS_AUTHENTICATION
+  );
+  const exportDirectoryPath = castVoteRecordExport.asDirectoryPath();
+  fs.appendFileSync(
+    path.join(exportDirectoryPath, CastVoteRecordExportFileName.METADATA),
+    '}'
+  );
+
+  expect(await readCastVoteRecordExport(exportDirectoryPath)).toEqual(
+    err({ type: 'metadata-file-parse-error' })
+  );
+});
+
+test.each<{ modifier: (castVoteRecordReportPath: string) => void }>([
+  {
+    modifier: (castVoteRecordReportPath) =>
+      fs.appendFileSync(castVoteRecordReportPath, '}'),
+  },
+  {
+    modifier: (castVoteRecordReportPath) =>
+      fs.writeFileSync(castVoteRecordReportPath, JSON.stringify({})),
+  },
+  {
+    modifier: (castVoteRecordReportPath) =>
+      fs.writeFileSync(castVoteRecordReportPath, JSON.stringify({ CVR: [] })),
+  },
+])('cast vote record parse error during import', async ({ modifier }) => {
+  mockFeatureFlagger.enableFeatureFlag(
+    BooleanEnvironmentVariableName.SKIP_CAST_VOTE_RECORDS_AUTHENTICATION
+  );
+  const exportDirectoryPath = await modifyCastVoteRecordExport(
+    castVoteRecordExport.asDirectoryPath(),
+    { numCastVoteRecordsToKeep: 1 }
+  );
+  const castVoteRecordExportSubDirectoryNames =
+    await getCastVoteRecordExportSubDirectoryNames(exportDirectoryPath);
+  expect(castVoteRecordExportSubDirectoryNames).toHaveLength(1);
+  const castVoteRecordId = assertDefined(
+    castVoteRecordExportSubDirectoryNames[0]
+  );
+  const castVoteRecordReportPath = path.join(
+    exportDirectoryPath,
+    castVoteRecordId,
+    CastVoteRecordExportFileName.CAST_VOTE_RECORD_REPORT
+  );
+  modifier(castVoteRecordReportPath);
+
+  const { castVoteRecordIterator } = (
+    await readCastVoteRecordExport(exportDirectoryPath)
+  ).unsafeUnwrap();
+  const castVoteRecordResults = await castVoteRecordIterator.toArray();
+  expect(castVoteRecordResults).toHaveLength(1);
+  const castVoteRecordResult = assertDefined(castVoteRecordResults[0]);
+  expect(castVoteRecordResult).toEqual(
+    err({ type: 'invalid-cast-vote-record', subType: 'parse-error' })
+  );
```

Comment:
Simplify to this?
```suggestion
  expect(castVoteRecordResults).toEqual([
    err({ type: 'invalid-cast-vote-record', subType: 'parse-error' })
  ]);
```

Reply:
:100: https://github.com/votingworks/vxsuite/pull/4091/commits/d4ed6dac3986f6285e0469fca2cede3ccbc1a215